shopping_mall/backend/ai/rag.py

```
@@ -322,6 +351,58 @@ def retrieve_with_scores(
+            }
+            if where:
+                kwargs["where"] = where
+
+            results = col.query(**kwargs)
+            documents = results.get("documents", [[]])[0]
+            distances = results.get("distances", [[]])[0]
+            metadatas = results.get("metadatas", [[]])[0]
+
+            filtered = [
+                (doc, meta or {})
+                for doc, dist, meta in zip(documents, distances, metadatas)
```

Ruff (0.15.12)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)